test/integration/BCowPool.t.sol

```diff
@@ -0,0 +1,112 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.25;
+
+import {IERC20} from '@cowprotocol/interfaces/IERC20.sol';
+import {GPv2Order} from '@cowprotocol/libraries/GPv2Order.sol';
+import {BCoWPool} from 'contracts/BCoWPool.sol';
+
+import {BConst} from 'contracts/BConst.sol';
+import {BNum} from 'contracts/BNum.sol';
+import {Test} from 'forge-std/Test.sol';
+import {IBPool} from 'interfaces/IBPool.sol';
+import {ISettlement} from 'interfaces/ISettlement.sol';
+
+// TODO: add GasSnapshot
+contract BCowPoolIntegrationTest is Test, BConst, BNum {
+  BCoWPool public pool;
+
+  IERC20 public dai = IERC20(0x6B175474E89094C44Da98b954EedeAC495271d0F);
+  IERC20 public weth = IERC20(0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2);
+  ISettlement public settlement = ISettlement(0x9008D19f58AAbD9eD0D60971565AA8510560ab41);
+
+  address public controller = makeAddr('controller');
+  address public lp = makeAddr('lp');
+  address public swapper = makeAddr('swapper');
+  address public solver = makeAddr('solver');
+
+  bytes32 public constant APP_DATA_HASH = bytes32(keccak256('exampleIntegrationAppData'));
+
+  function setUp() public {
+    vm.createSelectFork('mainnet', 12_593_265);
+
+    // deal controller
+    deal(address(dai), controller, 100e18);
+    deal(address(weth), controller, 100e18);
+
+    // deal LP
+    deal(address(dai), address(lp), 100e18);
+    deal(address(weth), address(lp), 100e18);
+
+    // deal swapper
+    deal(address(weth), address(swapper), 1e18);
+
+    vm.startPrank(controller);
+    // deploy
+    pool = new BCoWPool(address(settlement));
+    // enable trading
+    pool.enableTrading(APP_DATA_HASH);
```

Comment:
hmm, interesting, this is not supposed to happen, perhaps we could `enableTrading` only after `finalize` was called

Reply:
Will move it after finalize and we can add a revert inside `enableTrading` later